webapp/app/js/controllers/main.js

```diff
@@ -150,6 +149,16 @@ treeherder.controller('MainCtrl', [
                 }
             });
 
+            // Shortcut: enter a custom job or platform filter
+            Mousetrap.bind('f', function(ev) {
+                // Prevent shortcut key overflow during focus
+                ev.preventDefault();
+
+                $scope.$evalAsync(
+                    $rootScope.$broadcast('click-this', "platform-job-text-search-field")
```

Comment:
Totally agreed on this approach. I had searched for an S.O. on this and also dug through the angular docs but had missed the method. It's one of those that surely had to be there.